### PR TITLE
Added REST server to receive and process an image, and return a list …

### PR DESCRIPTION
**Pull request by guinnesschen**
…of detected objects

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -11,13 +12,17 @@
...
+def main(encodedImage):
+    image_64_decode = base64.decodebytes(encodedImage.encode('utf-8'))
 
-def main():
-    image = cv2.imread(imagefile)
+    with open('image.png', 'wb') as image_result:
+        image_result.write(image_64_decode)
+
+    image = cv2.imread('image.png')
+    os.remove('image.png')
```
Comment (SoloSynth1) [inline main.py:25]: @guinnesschen Any reason why writing the image into a file, then to read it again?

File: main.py [hunk 2]
```diff
@@ -31,28 +36,18 @@ def main():
...
                 class_ids.append(class_id)
+
     result = []
     for class_id in class_ids:
         result.append(labels[class_id])
-    # print(result)
-    new_filename = "objects_in_" + imagefile.split('.')[0] + ".txt"
-    with open(new_filename, "w") as f:
-        f.write(str(result))
+    return json.dumps(result)
```
Comment (SoloSynth1) [inline main.py]: @guinnesschen Please use `json.jsonify()` in `flask` framework

File: requirements.txt [hunk 3]
```diff
@@ -1,2 +1,3 @@
 numpy==1.19.1
 opencv-python==4.3.0.36
+flask
```
Comment (SoloSynth1) [inline requirements.txt]: @guinnesschen Please include version number, you can generate this dependency list in your shell by running `pip freeze > requirements.txt`.

File: server.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from flask import Flask, request
+from main import main
+
+app = Flask(__name__)
+
+@app.route('/', methods = ['POST'])
+def process_image():
+    return main(request.get_json()["data"])
```
Comment (SoloSynth1) [inline server.py]: @guinnesschen `main.main()` not a helpful name, especially when it's not the "main" entry-point of the API

File: server.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+if __name__ == "__main__":
+    app.run()
```
Comment (SoloSynth1) [inline server.py]: @guinnesschen Please add `host="0.0.0.0"`

**Review by SoloSynth1 — CHANGES_REQUESTED** [review]
@guinnesschen Please review the comments. Thanks!

File: main.py (hunk 1 above)
Comment (guinnesschen) [reply main.py:25]: cv3.imread reads from a file, and I wasn't sure if the return type of cv2.imread is the same as image_result. Can I skip this step?

**Pull request merged**